tools/calculate_ref_alt_counts.py

```diff
@@ -0,0 +1,153 @@
+                ref, alt, ign, mul, ref_sequence, alt_sequence = calculate_ref_count(
+                    bpp, read_length, self.reference_genome, bam, self.buffer)
+                log(bpp, name)
+                log('Calculated counts: Ref: {}, Alt: {}, Mul: {}, Ignored: {} '.format(
+                    len(ref), len(alt), len(mul), len(ign)))
+                log('Ref_probe: {}, Alt_probe: {}'.format(ref_sequence, alt_sequence))
+                info = {'{}_ref_count'.format(name): len(ref),
+                        '{}_alt_count'.format(name): len(alt),
+                        '{}_ignored_count'.format(name): len(ign)}
+                for key, value in info.items():
+                    data[key] = value
+                self.bpp_cache[bpp] = data
+        for key, value in self.bpp_cache[bpp].items():
+            bpp.data[key] = value
+        return bpp
+
+    def calculate_all_counts(self, input_files, output_file):
+        """
+        Helper method to calculate the ref and alt counts for all bpps in a file
+
+        Args:
+            input_files (list): List of mavis formatted files to use as input
+            output_file (str): Path to the desired output file
+        """
+        processed_bpps = {}
+        filtered_events = []
+
+        bpps = read_inputs(input_files, add_default={'stranded': False})
+
+        for bpp in bpps:
+            # only use precise bpps that are within a certain event size
+            try:
+                processed_bpps[bpp.product_id] = self.calculate_ref_counts(bpp)
+            except ValueError:
+                # wrong event type to calculate a ref/alt count
+                filtered_events.append(bpp)
+                continue
+
+        log('filtered {} events'.format(len(filtered_events)))
+
+        output_tabbed_file(processed_bpps.values(), output_file)
+        return processed_bpps, filtered_events
+
+
+def parse_arguments():
+    """
+    parse command line arguments
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '-o', '--output',
+        help='path to the output file', required=True, metavar='FILEPATH')
+    parser.add_argument(
+        '-n', '--input', required=True, metavar='FILEPATH', nargs='+',
+        help='Path to the Input mavis summary file')
+    parser.add_argument(
+        '-b', '--bam', action='append', nargs=2, default=[],
+        help='name for the library and the path to its bam file', required=True, metavar=('<name>', 'FILEPATH'))
+    parser.add_argument(
+        '-r', '--reference', required=True, metavar='FILEPATH',
+        help='Path to the Input reference genome fasta file')
+    args = parser.parse_args()
+    return args
+
+
+def main():
+    """
+    main entry point
+    """
+    log_conf = {'format': '{message}', 'style': '{', 'level': 1}
+    logging.basicConfig(**log_conf)
+    args = parse_arguments()
+    calculate_all_counts(args.reference_genome, args.bam, args.input, args.output_file)
```

Comment:
how are you able to call this instance method without instantiating the class? does this run? is it some python magic I've not heard of?

Reply:
I just forgot to instantiate the class here.